UefiCpuPkg/Library/RegisterCpuFeaturesLib/CpuFeaturesInitialize.c

```diff
@@ -130,7 +130,13 @@ CpuInitDataInitialize (
   GetNumberOfProcessor (&NumberOfCpus, &NumberOfEnabledProcessors);
 
   CpuFeaturesData->InitOrder = AllocatePages (EFI_SIZE_TO_PAGES (sizeof (CPU_FEATURES_INIT_ORDER) * NumberOfCpus));
-  ASSERT (CpuFeaturesData->InitOrder != NULL);
+  // MU_CHANGE [BEGIN] - CodeQL change

```

Comment:
Looking at this again, there's a lot of allocations in the function. I'm planning to:

1. Initialize all of the pointers that get assigned memory to NULL
2. Route all cases where an allocation fails to a label at the end of the function that frees buffers and restores external pointers (like `CpuFeaturesData->InitOrder`) to a known value (e.g. `NULL`) on an allocation failure.

Reply:
Updated